### PR TITLE
Rails Capstone Project - Budget app

### PR DESCRIPTION
**Pull request by Hannziegel**
### Interactions

- [x]  Splash screen

  - A simple page with the name of your app (yes, you need to choose one), and links to the sign-up and login pages.

- [x]  Sign up and login pages
  - The user should be able to register in the app with full name, email, and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- [x] Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon, and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- [x] Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page, the user can see the total amount for the category (the sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- [x]  "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- [x] "Add a new transaction" page
  - The user fills out a form to create a new transaction with the following:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

File: app/views/categories/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      <%= form.label :icon, style: "display: block" %>
+      <%= form.select :icon, [ ['🍺'], ['🍔'], ['💪'], ['🐺'], ['🐱'],
+                               ['✈'], ['💅'], ['🎮'], ['💊'], ['🩺'],
+                               ['💘'], ['💰'], ['💼'],[ '🎵'], ['🧦'] ]
+      %>
+    </div>
+
+    <%= form.submit class: "submit-btn" %>
+  <% end %>
+</div>
```
Comment (bimbolabuari) [inline app/views/categories/_form.html.erb:34]: While being able to create a new category and I think you missed a little requirement. 🤔 

 - "Add a new category" page

      - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
     - **The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success**.
     - When the user clicks on the "Back" button (<), the user navigates to the home page.
   
When I click on the save button, am not taken to the home page on success, but to a page, that shows that the category was successfully recorded( the category id route), however, when I then click the back button I am taken to the home page as required. 👍 

https://user-images.githubusercontent.com/66112760/205709192-ddd3d78e-1800-412e-8d63-f35c0273422e.mov


N.B: Please make this requested change also in creating new transactions(payment).

**Review by bimbolabuari — CHANGES_REQUESTED** [review]
_**This is a manual review**_

## Highlights:

- Your app works as expected. ✔️ 
- Usage of Postgres as your database. ✔️ 
- The project is deployed and accessible online. ✔️ 
-  Usage of devise for authentication. ✔️ 
- Nice video presentation  ✔️ 

## Request Changes 👍 
_Kindly check the inline comment_

Cheers and Happy coding!:clap::clap::clap:

File: app/views/categories/_form.html.erb (hunk 1 above)
Comment (Hannziegel) [reply app/views/categories/_form.html.erb:34]: Hi @bimbolabuari , thank you for your review. I did the changes as kindly required in both, the new category, and the new transaction. Now is working as expected. 😃

**Review by bimbolabuari — COMMENTED** [review]
_**This is a manual review**_

## Highlights:

- Implemented requested change ✔️ 
  - 

https://user-images.githubusercontent.com/66112760/205729960-ab18dc2c-2d98-4eb8-88d6-db49b8717172.mov



  ## Approved 🥳 

Note: It will be approved on your dashboard as soon as possible.

Cheers and Happy coding!:clap::clap::clap:

**Pull request merged**